lib/wos_queries.rb

```diff
@@ -48,6 +48,19 @@ def retrieve_by_id(uids)
     retrieve_records(:retrieve_by_id, message)
   end
 
+  # @param doi [String] a digital object identifier (DOI)
+  # @return [WosRecords]
+  def search_by_doi(doi)
+    message = search_by_doi_params(doi)
+    response = wos_client.search.call(:search, message: message)
+    records = records(response, :search_response)
+    # Return a unique DOI match or nothing, because the WoS API does partial string matching
+    # on the `DO` field.  When the result set is only one record, it's likely to be a good match; but
+    # otherwise the results could be nonsense.
+    return records if records.count == 1
```

Comment:
just reminding myself -- this return will prevent the following line from executing when reached, correct?

Reply:
Yes, when `records.count == 1` it returns `records` and short-circuits the rest of the block.